src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -505,6 +566,23 @@ namespace syrec {
                 }
             }
         }
+    }
+
+    auto DDSynthesizer::synthesize(dd::mEdge src, TruthTable::CubeMap const& codeword, std::size_t const& r, std::size_t const& m, std::unique_ptr<dd::Package<>>& dd) -> qc::QuantumComputation& {
+        qc.clear();
+        runtime  = 0.;
+        numGates = 0U;
+
+        const auto start = std::chrono::steady_clock::now();
+
+        synthesizeDD(src, dd, m, false);
+
+        // if codeword is empty, then function is src edge realizes a reversible function by default.
```

Comment:
Please look through all the comments for typos and mistakes... CLion even has a language plugin for that.

Furthermore, the reasoning in this sentence is again messed up. `"// if codeword is empty"` followed by `if(!codeword.empty())`